dc/s2n-quic-dc/src/path/secret/map.rs

```diff
@@ -684,13 +684,15 @@ impl Entry {
         secret: schedule::Secret,
         sender: sender::State,
         receiver: receiver::State,
-        mut parameters: ApplicationParams,
+        parameters: ApplicationParams,
         rehandshake_time: Duration,
     ) -> Self {
         // clamp max datagram size to a well-known value
-        parameters.max_datagram_size = parameters
-            .max_datagram_size
-            .min(crate::stream::MAX_DATAGRAM_SIZE as _);
+        let max_datagram_size = parameters.max_datagram_size.load(Ordering::Relaxed);
+        parameters.max_datagram_size.store(
+            max_datagram_size.min(crate::stream::MAX_DATAGRAM_SIZE as _),
+            Ordering::Relaxed,
+        );
```

Comment:
This should use `fetch_min`: https://doc.rust-lang.org/std/sync/atomic/struct.AtomicU64.html#method.fetch_min

Reply:
good call, I could see Clippy catching this someday